crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -0,0 +1,911 @@
+/// ```python
+/// def func1():
+///     pass
+/// def func2():
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def func1():
+///     pass
+///
+///
+/// def func2():
+///     pass
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E302.html)
+#[violation]
+pub struct BlankLinesTopLevel {
+    actual_blank_lines: u32,
+}
+
+impl AlwaysFixableViolation for BlankLinesTopLevel {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let BlankLinesTopLevel {
+            actual_blank_lines: nb_blank_lines,
+        } = self;
+
+        format!("Expected {BLANK_LINES_TOP_LEVEL:?} blank lines, found {nb_blank_lines}")
+    }
+
+    fn fix_title(&self) -> String {
+        "Add missing blank line(s)".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks for extraneous blank lines.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends using blank lines as follows:
+/// - No more than two blank lines between top-level statements.
+/// - No more than one blank line between non-top-level statements.
+///
+/// ## Example
+/// ```python
+/// def func1():
+///     pass
+///
+///
+///
+/// def func2():
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def func1():
+///     pass
+///
+///
+/// def func2():
+///     pass
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E303.html)
+#[violation]
+pub struct TooManyBlankLines {
+    actual_blank_lines: u32,
+}
+
+impl AlwaysFixableViolation for TooManyBlankLines {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyBlankLines {
+            actual_blank_lines: nb_blank_lines,
+        } = self;
+        format!("Too many blank lines ({nb_blank_lines})")
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove extraneous blank line(s)".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks for missing blank line after function decorator.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends the use of blank lines as follows:
+/// - Two blank lines are expected between functions and classes
+/// - One blank line is expected between methods of a class.
+///
+/// ## Example
+/// ```python
+/// class User(object):
+///
+///     @property
+///
+///     def name(self):
+///         pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class User(object):
+///
+///     @property
+///     def name(self):
+///         pass
+/// ```
+///
+/// ## References
+/// - [PEP 8](https://peps.python.org/pep-0008/#blank-lines)
+/// - [Flake 8 rule](https://www.flake8rules.com/rules/E304.html)
+#[violation]
+pub struct BlankLineAfterDecorator;
+
+impl AlwaysFixableViolation for BlankLineAfterDecorator {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("blank lines found after function decorator")
```

Comment:
We should capitalize these for consistency with the rest

Reply:
Done in fdf603542